### DISCUSSION TITLE
Being able to set a model value to None during validation

### DISCUSSION THREAD

**Discussion by Colin-b** [pydantic/pydantic#9739] [Feature Request]
Hello,

Based on the content of some fields, I would like to discard the full model without raising a validation error.
Any attribute with my model as type should then default to None in such a case (if every attribute is matching the condition of course).

It looks like I cannot return `None` on a `before` model validator.
And I want to keep this behavior within my model definition, I don't want to replicate it per model with attributes of my model as type.
Returning an empty dict does not match my use case as it would trigger the validation of every mandatory field, and that's not what I want.

It is possible in the current state of pydantic? And if not, is it something that could be foreseen?

Thanks again

**Comment by orfisko** [comment]
If I understand your question correctly, I think you need to solve it in the model that references your model instead of making the model return itself as None. Perhaps wrapping the concerned model with Annotated and referencing the annotated type could solve this issue if you still want to implement it as such?

**Reply by Colin-b** [reply]
In the end I went for returning None in "after" model validator, and have additional "after" model validator checking the values on the models referencing it to do some custom behavior.
